CanvasPlusPlayground/PeopleManager.swift

```diff
@@ -58,23 +58,26 @@ class PeopleManager {
         }
     }
     
-    func fetchActiveCourses() async {
+    private func fetchActiveCourses() async -> [Course] {
         guard let (data, _) = try? await CanvasService.shared.fetchResponse(.getCourses(enrollmentState: "active")) else {
-            print("Failed to fetch files.")
-            return
+            print("Failed to fetch courses.")
+            return []
         }
         
         if let retCourses = try? JSONDecoder().decode([Course].self, from: data) {
-            self.courses = retCourses
+            return retCourses
         } else {
             print("Failed to decode file data.")
+            return []
         }
     }
     
-    func fetchAllClassesWith(userID: Int) async -> ([Course]) {
-        await fetchActiveCourses()
-        
-        var commonCourses = [Course]()
+    func fetchAllClassesWith(
+        userID: Int,
+        receivedNewCourse: @escaping (Course) -> Void = { _ in }
+    ) async {
+        let courses = await fetchActiveCourses()
```

Comment:
btw why are we fetching courses again, we already did that in course manager so we might as well pass it, fetch activeCourses here is doing another api call